### ISSUE TITLE
memory error while running autosklearn

### ISSUE THREAD

**Issue by ushmalramesh** [automl/auto-sklearn#662]
I'm getting memory error with pickle in load_models while running auto-sklearn in parallel. The stack trace is as follows:

Traceback (most recent call last):
File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
self.run()
File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/multiprocessing/process.py", line 93, in run
self._target(*self._args, **self._kwargs)
File "/home/ushmal/PycharmProjects/automl/automl_parallel_sparse.py", line 72, in spawn_classifier
automl.fit(X_train, y_train, dataset_name=dataset_name)
File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/estimators.py", line 664, in fit
dataset_name=dataset_name,
File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/estimators.py", line 337, in fit
self._automl[0].fit(**kwargs)
File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/automl.py", line 996, in fit
load_models=load_models,
File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/automl.py", line 208, in fit
only_return_configuration_space=only_return_configuration_space,
File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/automl.py", line 370, in _fit
data_manager_path = self._backend.save_datamanager(datamanager)
File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/util/backend.py", line 318, in save_datamanager
pickle.dump(datamanager, fh, -1)
MemoryError

I'm using a sparse array input to train the models.

My train dataset has shape (117713,9570). The DataFrame has SparseSeries columns.

System Info:
Memory: 31.4 GB
Processor : AMD Ryzen 7 2700X Eight-Core Processor × 16 

I tried increasing ml_memory _limit but the error persisted.

**Comment by ushmalramesh** [comment]
@mfeurer Thanks for the useful pointers! I will try using scipy sparse matrices. It does appear that the dataset is too large and that's why the pickle throws a memory error. I did not conclude that the error occurs in load_models, just that memory error occurs during pickling. I will try performing actions you suggested and raise a new issue if the errors persist. Closing the issue for now.
